src/components/MaterialUI/Inputs/Input.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import InputMUI from "@material-ui/core/Input";
+import InputProps, { isInputProps } from "./InputProps";
+
+export const useStyles = makeStyles(theme => ({
+	container: {
+		display: "flex",
+	},
+	prepend: {
+		fontSize: theme.spacing(1.3),
+		lineHeight: theme.spacing(1.3),
+		padding: theme.spacing(0.85),
+		minWidth: theme.spacing(6),
+		marginRight: theme.spacing(-0.1),
+		height: theme.spacing(3),
+		display: "inline-flex",
+		borderRadius: theme.spacing(0.5, 0, 0, 0.5),
+		border: `${theme.spacing(0.1)} solid ${theme.palette.grey.borders}`,
+		boxSizing: "border-box",
+		backgroundColor: theme.palette.grey.light,
+		whiteSpace: "nowrap",
+	},
+	controlInput: {
+		height: theme.spacing(1.6),
+		maxWidth: "unset",
+		border: `${theme.spacing(0.1)} solid ${theme.palette.grey.borders}`,
+		borderRadius: props => (props.label ? theme.spacing(0, 0.5, 0.5, 0) : theme.spacing(0.5)),
+		paddingLeft: theme.spacing(0.85),
+		"&:active": {
+			borderRadius: props => (props.label ? theme.spacing(0, 0.5, 0.5, 0) : theme.spacing(0.5)),
+		},
+		"&:focus": {
+			border: `${theme.spacing(0.1)} solid ${theme.palette.focus}`,
+			borderRadius: props => (props.label ? theme.spacing(0, 0.5, 0.5, 0) : theme.spacing(0.5)),
+		},
+	},
+}));
+
+const Input = ({ inputProps }) => {
```

Comment:
I don't think this is good name. KeyValueField or KeyValueInput tells me much more about what is it. Input is generic name and Input inside Inputs folder makes me to think that it's like a parent object for all other inputs.

Reply:
MUI also have Input component
The key is just an option, you can choose not to display it